sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JacksonGenerator.scala

```diff
@@ -118,7 +36,7 @@ private[sql] object JacksonGenerator {
    * @param gen a JsonGenerator object
    * @param row The row to convert
    */
-  def apply(rowSchema: StructType, gen: JsonGenerator, row: InternalRow): Unit = {
+  def apply(rowSchema: StructType, gen: JsonGenerator)(row: InternalRow): Unit = {
```

Comment:
The removed `apply` method has the same signature. And personally I think this one is semantically more preferable: `JacksonGenerator(schema, gen)` returns a generator, then we apply this generator with a row to generate actual JSON records.